plugins/system/redshop/redshop.php

```diff
@@ -75,4 +77,80 @@ public function onAfterInitialise()
             $session->set('product_currency', $newCurrencyId);
         }
     }
+    
+    /**
+     * After Display Product method
+     *
+     * Method is called by the product view
+     *
+     * @param   string  &$template  The Product Template Data
+     * @param   object  $params     The product params
+     * @param   object  $data       The product object Data
+     *
+     * @return  void
+     */
+    public function onAfterDisplayProduct(&$template, $params, $data)
+    {
+        $app    = \JFactory::getApplication();
+        $url    = Uri::getInstance()->toString();
+        $option = $app->input->get('option', '');
+        $view   = $app->input->get('view', '');
+        
+        if (!$app->isClient('site'))
+        {
+            return;
+        }
+        
+        $vatprice              = RedshopHelperProduct::getProductTax($data->product_id, $data->product_price);
+        $currencySymbol        = Redshop::getConfig()->get('REDCURRENCY_SYMBOL');
+        $isStock               = RedshopHelperStockroom::isStockExists($data->product_id);
+        $getBrand              = null;
+        $discountPrice         = $data->discount_price + $vatprice;
+        $normalPrice           = $data->product_price + $vatprice;
+
+        $getProductDesc        = !empty($data->product_s_desc) ? $data->product_s_desc : $data->product_desc;
+
+        $product_desc_clean    = str_replace(array( '\'', '"'), "", strip_tags($getProductDesc));
+        $productNumber         = strip_tags($data->product_number);
+        $getConfigUseStockRoom = Redshop::getConfig()->get('USE_STOCKROOM');
+        $path                  = (REDSHOP_FRONT_IMAGES_ABSPATH . 'product/');
+
+        $price                 = $data->product_on_sale ? $discountPrice : $normalPrice;
+        $getStockroom          = $isStock ? 'http://schema.org/InStock' : 'https://schema.org/OutOfStock';
+        
+        if (isset($data->manufacturer_name))
+        {
+            $getBrand = '"brand": {
+                     "@type": "Brand",
+                     "name": "'.$data->manufacturer_name.'"
+                     },';
+        }
+        
+        if ($getConfigUseStockRoom && $option == 'com_redshop' && $view == 'product' )
```

Comment:
why you added `$getConfigUseStockRoom`  into if here ?  if admin user does not enable stock feature, it causes the json schema does not load.